components/eamxx/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -1378,12 +1378,50 @@ T get_attribute (const std::string& filename,
   if (varname=="GLOBAL") {
     varid = PIO_GLOBAL;
   } else {
-    varid = impl::get_var(filename,varname,"scorpio::set_any_attribute").ncid;
+    varid = impl::get_var(filename,varname,"scorpio::get_attribute").ncid;
   }
 
+  // If the attribute type does not match T, we need a temporary, since we can't pass T* where pio expects
```

Comment:
This was the real bug. Basically, when the var has float dtype, also the FillValue has float dtype, but we were trying to access it passing a double pointer. When PIO does a reinterpret_cast of the input pointer to a float* and writes to it, it only writes the first 32bits (and the meaning of those 32 bits is not the same as in the full double), leaving some bits uninitialized. In fact, reading the fillvalue gave a completely random number (like 1e-300), rather than the expeted 1e30 (or whatever we had in the file).